docs/generate-and-template.md

```diff
@@ -173,10 +173,11 @@ However, environment variables are expanded.
 Along with any existing environment variables, the following variables
 are specially defined:
 
-- HELM_HOME: The Helm home directory
-- HELM_DEFAULT_REPO: The repository alias for the default repository.
-- HELM_GENERATE_FILE: The present file's name
-- HELM_GENERATE_DIR: The absolute path to the chart directory of the present chart
+- `HELM_HOME`: The Helm Classic home directory
+    - Although the value of the `HELMC_HOME` environment variable, if defined, may play a role in defining the specially expanded `HELM_HOME` variable, the two are not equivalent. To produce charts that remain compatible with the _original_ Helm tool, which has now become Helm Classic, only the `HELM_HOME` variable should be used within `helm:generate` headers.
```

Comment:
@mboersma Ya.  It's confusing.  :(  The helmc binary itself is (or rather, _will be_) sensitive to the `HELMC_HOME` environment variable, which can _optionally_ be used to specify a custom home (rather than the default).

_However..._ to be backwards compatible with generator directives that appear in _existing_ charts (created for the _original_ helm), `HELM_HOME` (no `C`) will continue to be expanded there as it always has.

If someone opted to use `HELMC_HOME` in a generator directive, that chart wouldn't be usable by people using an older version of helm because the older version does not expand that variable.

I know that's all confusing.  fwiw, the _new_ (kubernetes/)helm wants to use `HELM_HOME` defaulted to `~/.helm`, which is why that team had +1'ed helm classic having its own distinct home-- so both tools can co-exist for some period of time.

I'm not sure how to make all of this clear in the docs without being as verbose as I just was.  :(  I am open to all suggestions on rewording.